sdk/core/core/src/az_json_private.h

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef _az_JSON_PRIVATE_H
+#define _az_JSON_PRIVATE_H
+
+#include <az_json.h>
+#include <az_precondition_internal.h>
+
+#include <_az_cfg_prefix.h>
+
+enum
+{
+  // We are using a uint64_t to represent our nested state, so we can only go 64 levels deep.
+  _az_MAX_JSON_STACK_SIZE = sizeof(uint64_t) * 8 // 64
```

Comment:
This is fragile.  Use the sizeof the field in json_stack.  That way if the field in the structure changes this value changes as well.